rs/execution_environment/src/canister_manager.rs

```diff
@@ -864,12 +865,15 @@ impl CanisterManager {
         );
         match dts_result {
             DtsInstallCodeResult::Finished {
-                canister,
+                mut canister,
                 call_id: _,
                 message: _,
                 instructions_used,
                 result,
-            } => (result, instructions_used, Some(canister)),
+            } => {
+                canister.update_on_low_wasm_memory_hook_condition();
```

Comment:
This is required to mimic the behavior of `execute_install_code` because `execute_install_code` updates hook status in the `process_install_code` result. The reason is that `install_code` is only used in tests, and should behave similarly.